crates/matrix-sdk/src/client/mod.rs

```diff
@@ -1991,7 +1991,7 @@ impl Client {
         if let Some(session) = self.session() {
             match session {
                 AuthSession::Matrix(s) => {
-                    client.matrix_auth().inherit_session(s).await?;
+                    client.matrix_auth().restore_session(s).await?;
```

Comment:
Could we just have it call `Client::restore_session()` instead of matching for each session?

Reply:
Indeed, thanks!